core/gui/src/app/workspace/service/ai-assistant/ai-assistant.service.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable } from "@angular/core";
+import { firstValueFrom } from "rxjs";
+import { AppSettings } from "../../../common/app-setting";
+import { HttpClient, HttpHeaders } from "@angular/common/http";
+import { Observable, of } from "rxjs";
+import { map, catchError } from "rxjs/operators";
+
+// The type annotation return from the LLM
+export type TypeAnnotationResponse = {
+  choices: ReadonlyArray<{
+    message: {
+      content: string;
+    };
+  }>;
+};
+
+export const AI_ASSISTANT_API_BASE_URL = `${AppSettings.getApiEndpoint()}/aiassistant`;
+
+@Injectable({
+  providedIn: "root",
+})
+export class AIAssistantService {
+  constructor(private http: HttpClient) {}
+
+  public checkAIAssistantEnabled(): Observable<string> {
```

Comment:
please add comments for this function. what are the possible "string" values to return? 

You may also define 

```
const string AI_MODEL = "OpenAI" | "NoAiAssistant";
public checkAIAssistantEnabled(): Observable<AI_MODEL> {

}
```
to guard the only possible return values.